deployment/common/changeset/set_config_mcms_test.go

```diff
@@ -0,0 +1,328 @@
+package changeset_test
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	chain_selectors "github.com/smartcontractkit/chain-selectors"
+	"github.com/smartcontractkit/chainlink-common/pkg/utils/tests"
+
+	commonchangeset "github.com/smartcontractkit/chainlink/deployment/common/changeset"
+	"github.com/smartcontractkit/chainlink/deployment/common/proposalutils"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+
+	"github.com/smartcontractkit/chainlink/deployment/common/types"
+	"github.com/smartcontractkit/chainlink/deployment/environment/memory"
+)
+
+// setupSetConfigTestEnv deploys all required contracts for the setConfig MCMS contract call.
+func setupSetConfigTestEnv(t *testing.T) deployment.Environment {
+
+	lggr := logger.TestLogger(t)
+	cfg := memory.MemoryEnvironmentConfig{
+		Nodes:  1,
+		Chains: 2,
+	}
+	env := memory.NewMemoryEnvironment(t, lggr, zapcore.DebugLevel, cfg)
+	chainSelector := env.AllChainSelectors()[0]
+	config := proposalutils.SingleGroupMCMS(t)
+
+	// Deploy MCMS and Timelock
+	env, err := commonchangeset.ApplyChangesets(t, env, nil, []commonchangeset.ChangesetApplication{
+		{
+			Changeset: commonchangeset.WrapChangeSet(commonchangeset.DeployLinkToken),
+			Config:    []uint64{chainSelector},
+		},
+		{
+			Changeset: commonchangeset.WrapChangeSet(commonchangeset.DeployMCMSWithTimelock),
+			Config: map[uint64]types.MCMSWithTimelockConfig{
+				chainSelector: {
+					Canceller:        config,
+					Bypasser:         config,
+					Proposer:         config,
+					TimelockMinDelay: big.NewInt(0),
+				},
+			},
+		},
+	})
+	require.NoError(t, err)
+	return env
+}
+
+// TestSetConfigMCMS tests the SetConfigMCMS changeset by calling SetConfig and checking the config values.
+func TestSetConfigMCMS(t *testing.T) {
+	t.Parallel()
+	ctx := tests.Context(t)
+
+	env := setupSetConfigTestEnv(t)
+	chainSelector := env.AllChainSelectors()[0]
+	chain := env.Chains[chainSelector]
+	addrs, err := env.ExistingAddresses.AddressesForChain(chainSelector)
+	require.NoError(t, err)
+	require.Len(t, addrs, 6)
+
+	mcmsState, err := commonchangeset.MaybeLoadMCMSWithTimelockChainState(chain, addrs)
+	require.NoError(t, err)
+
+	timelockAddress := mcmsState.Timelock.Address()
+	cfg := proposalutils.SingleGroupMCMS(t)
+	// Add the timelock as a signer to check state changes
+	cfg.Signers = append(cfg.Signers, timelockAddress)
+	cfg.Quorum = 2 // quorum should change to 2 out of 2 signers
+	// Transfer ownership to timelock
```

Comment:
I am a bit slow here, where do we transfer the ownership?

Reply:
I probably copied the structure from another test and missed removing this. Should be fixed now, sorry for the confusion https://github.com/smartcontractkit/chainlink/pull/15684/commits/d8cc495842e298e8a6b212fdb5c3bea4bd0929b5